### PR TITLE
Arch arm properly report stack corruption

### PR DESCRIPTION
**Pull request by ioannisg**
Allows ARM builds to report Stack Corruption.
Fixes issues reported in #7706

### CODE REVIEW

**Review by tyrantlizard77 — APPROVED** [review]
LGTM thanks!!

File: tests/kernel/fatal/src/main.c [hunk 1]
```diff
@@ -235,15 +235,9 @@ void test_fatal(void)
 			(k_thread_entry_t)stack_thread2,
 			NULL, NULL, NULL, K_PRIO_PREEMPT(PRIORITY), 0,
 			K_NO_WAIT);
-#ifdef CONFIG_ARM
-	/* FIXME: See #7706 */
-	zassert_true(crash_reason == _NANO_ERR_STACK_CHK_FAIL ||
-		     crash_reason == _NANO_ERR_HW_EXCEPTION, NULL);
```
Comment (MaureenHelm) [inline tests/kernel/fatal/src/main.c]: We still need the workaround for the k64 because the nxp mpu throws a bus fault, not an mpu fault.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by ioannisg** [comment]
You are right, I missed that. Only tested with Nordic SOCs. I will add back the workaround for NXP MPU only. 

I wonder if we could patch the Bus fault handler for NXP mpu, to identify relevant stack corruptions.

**Comment by ioannisg** [comment]
@MaureenHelm , I added back the workaround for NXP_MPU, please, double-check if this is ok.

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by ioannisg** [comment]
Re-based

**Review by carlescufi — APPROVED** [review]
Looks good

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**